site_scons/site_tools/go_builder.py

```diff
@@ -49,6 +49,20 @@ def _scan_go_file(node, env, _path):
     return includes
 
 
+def get_go_version(output):
+    """Capture only the number, ignoring optional rc"""
+    one_part = '([0-9]+)(?:rc[0-9]+)?'
+    return '.'.join(re.findall(fr'(?:go)?{one_part}\.{one_part}\.{one_part}', output)[0])
+
+
+def test_go():
+    """Quick unit test"""
+    assert "1.20.10" == get_go_version("go version go1.20.10 linux/amd64")
+    assert "1.2.3" == get_go_version("go version go1.2.3 Linux/amd64")
+    assert "1.20.10" == get_go_version("go version go1.20.10-daos linux/amd64")
+    assert "1.20.10" == get_go_version("go version go1.20rc2.10 linux/amd64")
```

Comment:
This is a pretty bizarre version, but I don't think it makes sense to contort ourselves into dealing with it. My proposed regex that only cares about the numbers and periods immediately after the ` go` string resolves this as `1.20`, and IMO this is good enough for the purposes of this check.

I changed that testcase and added a couple of others:
```suggestion
assert "1.20" == get_go_version("go version go1.20rc2.10 linux/amd64")
assert "1.22" == get_go_version("go version go1.22-20240109-RC01 cl/597041403 +dcbe772469 X:fieldtrack,boringcrypto linux/amd64")
assert "" == get_go_version("go version goquack-moo linux/amd64")
```